components/scripts/knowledge-dropdown.tsx

```diff
@@ -66,40 +99,114 @@ const ScriptKnowledgeDropdown = () => {
       {program && (
         <DropdownMenu
           onAction={(key) => {
-            removeFile(key as string);
+            if (key !== 'show-all') {
+              removeFile(key as string);
+            }
           }}
         >
-          {knowledgeFiles && knowledgeFiles.length ? (
-            <DropdownSection
-              title={
-                'Thread Knowledge' +
-                (knowledgeFiles.length > 0 ? ' (Click to remove)' : '')
-              }
-            >
-              {knowledgeFiles.map((f) => (
+          <DropdownSection
```

Comment:
Most of this is getting the counts displayed correctly and adding the "show all" functionality when there is more than 10 files.

Reply:
I don't know if we should add max height and make it scrollable. Show like 100 files will still be too long if user clicked show all,

Reply:
Not sure what you are suggesting here. The max height and scroll works for lots of files.

Reply:
I am suggesting adding a max height and make it scrollable. Show all will still mess up with dropdown if there are a lot of files.

Reply:
Oh, cool. That is already here. So we're good!

Reply:
Here is an example.

<img width="942" alt="Screenshot 2024-08-30 at 07 43 36" src="https://github.com/user-attachments/assets/1ebe46ab-5d0e-49a5-8854-7413369eeea4">

<img width="935" alt="Screenshot 2024-08-30 at 07 43 59" src="https://github.com/user-attachments/assets/66cd98d7-2b54-4f59-acb2-e66002dd4dc8">

After I took those screenshots, I made the "Show All" text smaller.
<img width="933" alt="Screenshot 2024-08-30 at 07 47 56" src="https://github.com/user-attachments/assets/b4b1511e-2db7-49e3-8a28-a5ec65a5b80e">